jerry-core/ecma/base/ecma-globals.h

```diff
@@ -204,6 +204,16 @@ typedef struct
   void *info_p; /**< free info or callback */
 } ecma_native_pointer_t;
 
+#ifdef JERRY_VM_EXEC_STOP
+
+/**
+ * Callback which returns with a non-undefined value
+ * when the ECMAScript execution should be stopped.
+ */
+typedef ecma_value_t (*jerry_vm_exec_stop_cb_t) (void *user_p);
```

Comment:
This is not bad news, this is how existing callback types are already used.
- jerry_external_handler_t in ecma-function-object.c
- jerry_object_free_callback_t in ecma-gc.c
- jerry_user_context_deinit_cb in jcontext.h

They are all defined in jerryscript.h and used in the engine. There is no "dependency to be removed". It's the callback typedef duplication that's to be removed from ecma-globals.h.